cli/compose/interpolation/interpolation.go

```diff
@@ -80,7 +76,7 @@ func recursiveInterpolate(
 	case []interface{}:
 		out := make([]interface{}, len(value))
 		for i, elem := range value {
-			interpolatedElem, err := recursiveInterpolate(elem, mapping)
+			interpolatedElem, err := recursiveInterpolate(elem, path, opts)
```

Comment:
I think maybe lists should have a `[]` item in the path to differentiate a path to a scalar or a path to an item in the list.